src/main/java/software/amazon/awssdk/crt/CrtResource.java

```diff
@@ -14,26 +14,41 @@
  */
 package software.amazon.awssdk.crt;
 
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.ConcurrentHashMap;
+
 /**
  * This wraps a native pointer to an AWS Common Runtime resource. It also ensures
  * that the first time a resource is referenced, the CRT will be loaded and bound.
  */
 public class CrtResource {
+    private static final ConcurrentHashMap<Long, String> NATIVE_RESOURCES = new ConcurrentHashMap<>();
     private long ptr;
 
     static {
         /* This will cause the JNI lib to be loaded the first time a CRT is created */
         new CRT();
     }
 
+    public static int getAllocatedNativeResourceCount() {
+        return NATIVE_RESOURCES.size();
+    }
+
+    public static Collection<String> getAllocatedNativeResources() {
```

Comment:
Are the strings useful in identifying the outstanding objects? It feels like you'd need the address too, especially if trying to correlate them to logs or something.

Reply:
This function helps find the Java CrtResource Class Names that haven't been released yet, and is useful for finding CrtResource leaks. I'm hesitant to expose the raw pointers through the global public API since then other Threads can use the global API to get pointers that don't belong to them and do things they shouldn't be allowed to do.